roles/tests-integration/tests/common/sniffer.rs

```diff
@@ -175,13 +225,39 @@ impl Sniffer {
         recv: Receiver<MessageFrame>,
         send: Sender<MessageFrame>,
         upstream_messages: MessagesAggregator,
+        interrupt_messages: Vec<InterruptMessage>,
     ) -> Result<(), SnifferError> {
         while let Ok(mut frame) = recv.recv().await {
             let (msg_type, msg) = Self::message_from_frame(&mut frame);
-            upstream_messages.add_message(msg_type, msg);
+            for interrupt_message in interrupt_messages.iter() {
+                if interrupt_message.direction == MessageDirection::ToDownstream
+                    && interrupt_message.expected_message_type == msg_type
```

Comment:
I wonder if only matching against `direction` + `msg_type` isn't too generic?

shouldn't we consider matching against `msg` itself?

that way, there's no room for "accidentally" interrupting on a message simply because it happened before the expected trigger while having the same message type